code/modules/surgery/organs/tongue.dm

```diff
@@ -92,6 +92,83 @@
+	var/obj/structure/statue/custom/statue
+
+/datum/action/item_action/organ_action/statue/New(Target)
+	. = ..()
+	statue = new(null)
+	RegisterSignal(statue, COMSIG_PARENT_QDELETING, .proc/statue_destroyed)
+
+/datum/action/item_action/organ_action/statue/Destroy()
+	UnregisterSignal(statue, COMSIG_PARENT_QDELETING)
+	QDEL_NULL(statue)
+	. = ..()
+
+/datum/action/item_action/organ_action/statue/Trigger()
+	. = ..()
+	if(!iscarbon(owner))
+		to_chat(owner, "<span class='warning'>Your body rejects the powers of the tongue!</span>")
+		return
+	var/mob/living/carbon/becoming_statue = owner
+	if(becoming_statue.health < 1)
+		to_chat(becoming_statue, "<span class='danger'>You are too weak to become a statue!</span>")
+		return
+	if(ability_cooldown >= world.time)
+		to_chat(becoming_statue, "<span class='warning'>You just used the ability, wait a little bit!</span>")
+		return
+	var/is_statue = becoming_statue.loc == statue
+	to_chat(becoming_statue, "<span class='notice'>You begin to [is_statue ? "break free from the statue" : "make a glorious pose as you become a statue"]!</span>")
+	if(!do_after(becoming_statue, (is_statue ? 5 : 30), target = get_turf(becoming_statue)))
+		to_chat(becoming_statue, "<span class='warning'>Your transformation is interrupted!</span>")
+		ability_cooldown = world.time + 3 SECONDS
+		return
+	ability_cooldown = world.time + 10 SECONDS
+
+	if(statue.name == initial(statue.name)) //statue has not been set up
+		statue.name = "statue of [becoming_statue.real_name]"
+		statue.desc = "statue depicting [becoming_statue.real_name]"
+		statue.set_custom_materials(list(/datum/material/silver=MINERAL_MATERIAL_AMOUNT*5))
+
+	if(is_statue)
+		statue.visible_message("<span class='danger'>[statue] becomes animated!</span>")
+		becoming_statue.forceMove(get_turf(statue))
+		statue.moveToNullspace()
+		UnregisterSignal(becoming_statue, COMSIG_MOVABLE_MOVED)
+	else
+		becoming_statue.visible_message("<span class='notice'>[becoming_statue] hardens into a silver statue.</span>", "<span class='notice'>You have become a silver statue!</span>")
+		statue.set_visuals(becoming_statue.appearance)
+		statue.forceMove(get_turf(becoming_statue))
+		becoming_statue.forceMove(statue)
+		statue.obj_integrity = becoming_statue.health
+		RegisterSignal(becoming_statue, COMSIG_MOVABLE_MOVED, .proc/human_left_statue)
+
+	//somehow they used an exploit/teleportation to leave statue, lets clean up
+/datum/action/item_action/organ_action/statue/proc/human_left_statue(atom/movable/mover, atom/oldloc, direction)
+	SIGNAL_HANDLER
+
+	statue.moveToNullspace()
+	UnregisterSignal(mover, COMSIG_MOVABLE_MOVED)
+
+/datum/action/item_action/organ_action/statue/proc/statue_destroyed(datum/source)
+	to_chat(owner, "<span class='userdanger'>Your existence as a living creature snaps as your statue form crumbles!</span>")
+	if(iscarbon(owner))
+		//drop everything, just in case
+		var/mob/living/carbon/dying_carbon = owner
+		for(var/obj/item/W in dying_carbon)
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names